### PR TITLE
[tests-only] e2eTest. Api step. create folder in personal space

### PR DESCRIPTION
**Pull request by ScharfViktor**
part of https://github.com/owncloud/web/issues/8101

- user creates folder in the personal space

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/31118/11/1
:boom: The e2e-oc10 tests pipeline failed. The build has been cancelled.

File: tests/e2e/cucumber/features/smoke/reshare.feature [hunk 1]
```diff
@@ -16,13 +16,12 @@ Feature: reshare
     And "Admin" adds user to the group
       | user  | group |
       | Carol | sales |
+    And "Alice" creates the following folder in the personal space using API
```
Comment (SagarGi) [inline tests/e2e/cucumber/features/smoke/reshare.feature:19]:
```suggestion
    And "Alice" creates the following folder in personal space using API
```

File: tests/e2e/cucumber/steps/api.ts [hunk 2]
```diff
@@ -100,3 +100,14 @@ Given(
     }
   }
 )
+
+Given(
+  '{string} creates the following folder(s) in the personal space using API',
```
Comment (SagarGi) [inline tests/e2e/cucumber/steps/api.ts:105]:
```suggestion
  '{string} creates the following folder(s) in personal space using API',
```

**Review by SagarGi — COMMENTED** [review]
Code  LGTM :+1:

File: tests/e2e/cucumber/steps/api.ts [hunk 3]
```diff
@@ -100,3 +100,14 @@ Given(
...
+  async function (this: World, stepUser: string, stepTable: DataTable): Promise<void> {
+    const user = this.usersEnvironment.getUser({ key: stepUser })
+    for (const info of stepTable.hashes()) {
+      console.log(info.name, 'info.name')
```
Comment (SwikritiT) [inline tests/e2e/cucumber/steps/api.ts:109]:
```suggestion
     
```
can be removed

File: tests/e2e/support/api/http.ts [hunk 4]
```diff
@@ -10,24 +10,21 @@ export const request = async ({
...
-    {
-      method,
-      body,
-      headers: {
-        'OCS-APIREQUEST': true as any,
-        ...(user && {
-          Authorization: 'Basic ' + Buffer.from(user.id + ':' + user.password).toString('base64')
-        })
-      }
+  return await fetch(join(config.backendUrl, path + (path.includes('?') ? '&' : '?')), {
```
Comment (saw-jan) [inline tests/e2e/support/api/http.ts:18]: I think, `format` param is required for oC10
```suggestion
  return await fetch(join(config.backendUrl, path + (path.includes('?') ? '&' : '?') + 'format=json'), {
```

**Comment by SagarGi** [comment]
@ScharfViktor https://github.com/owncloud/web/pull/8170 this PR cover up the creation of folder inside space project so. my be you may not need to implement it for space project.

File: tests/e2e/support/api/http.ts (hunk 4 above)
Comment (SagarGi) [reply tests/e2e/support/api/http.ts:18]: for the above request does not support query param format maybe if could do like this `join(config.backendUrl, path + (path.includes('?') ? '&' : '?') + ((!config.ocis) ? 'format=json' : ''))`? to work?

Comment (ScharfViktor) [reply tests/e2e/support/api/http.ts:18]: I did same: 
```
const formatJson = config.ocis ? '' : 'format=json'
  return await fetch(join(config.backendUrl, path + (path.includes('?') ? '&' : '?') + formatJson)
```

but when I create folder in oc10 I get error: 
`FetchError: invalid json response body at http://host.docker.internal:8080/login reason: Unexpected token < in JSON at position 0`

Comment (saw-jan) [reply tests/e2e/support/api/http.ts:18]: That might be because the MKCOL request returns an empty response body.

Comment (SagarGi) [reply tests/e2e/support/api/http.ts:18]: @ScharfViktor may be it because you are using `dav/spaces` endpoint for `oc10`

Comment (ScharfViktor) [reply tests/e2e/support/api/http.ts]: creating a folder in oc10 without `'format=json'` works, so I added `formatJson = false` to the `request` method.

**Review by saw-jan — COMMENTED** [review]
LGTM :+1:

**Comment by SagarGi** [comment]
@ScharfViktor can you merge this today if possible. i will be rebasing it for my PR.

**Review by SagarGi — APPROVED** [review]

**Pull request merged**